plugins/lsp/default.nix

```diff
@@ -245,18 +245,17 @@ lib.nixvim.plugins.mkNeovimPlugin {
           }"}
 
           for i, server in ipairs(__lspServers) do
-            if type(server) == "string" then
-              require("lspconfig")[server].setup(__setup)
-            else
-              local options = ${runWrappers cfg.setupWrappers "server.extraOptions"}
-
-              if options == nil then
-                options = __setup
-              else
-                options = vim.tbl_extend("keep", options, __setup)
-              end
+            local name = server
+            local options = nil
+            if type(name) ~= 'string' then
+              name = server.name
+              options = server.extraOptions
+            end
 
-              require("lspconfig")[server.name].setup(options)
+            vim.lsp.enable(name)
+            vim.lsp.config(name, __setup)
+            if options ~= nil then
```

Comment:
Ah my bad with `!=` vs `~=` (always trips me up).

In lua can we simplify to `if options then` or do we need to be explicit about which falsy cases we want to handle?